src/PubSubClient.cpp

```diff
@@ -225,11 +224,11 @@ bool PubSubClient::connect(const char* id, const char* user, const char* pass, c
 
 // reads a byte into result
 bool PubSubClient::readByte(uint8_t* result) {
-    uint32_t previousMillis = millis();
+    unsigned long previousMillis = millis();
     while (!_client->available()) {
         yield();
-        uint32_t currentMillis = millis();
-        if (currentMillis - previousMillis >= ((int32_t)this->socketTimeout * 1000)) {
+        unsigned long currentMillis = millis();
+        if (currentMillis - previousMillis >= ((unsigned long)this->socketTimeout * 1000)) {
```

Comment:
Why not `static_cast<unsigned long>(this->socketTimeout * 1000ul)` ?
`1000` is signed, `1000ul` is not.

Reply:
Thanks!